src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -233,14 +237,20 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_pr
     # Install azure-arc agents
     helm_install_release(chart_path, subscription_id, kubernetes_distro, resource_group_name, cluster_name,
                          location, onboarding_tenant_id, http_proxy, https_proxy, no_proxy, proxy_cert, private_key_pem, kube_config,
-                         kube_context, no_wait, values_file_provided, values_file)
+                         kube_context, no_wait, values_file_provided, values_file, azure_cloud)
 
     return put_cc_response
 
 
 def send_cloud_telemetry(cmd):
-    cloud_name = cmd.cli_ctx.cloud.name
-    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.AzureCloud': cloud_name})
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.AzureCloud': cmd.cli_ctx.cloud.name})
+    cloud_name = cmd.cli_ctx.cloud.name.upper()
+    # Setting cloud name to format that is understood by golang SDK.
+    if cmd.cli_ctx.cloud.endpoints.resource_manager == consts.PublicCloud_RMEndpoint:
+        cloud_name = consts.Azure_PublicCloudName
+    elif cmd.cli_ctx.cloud.endpoints.resource_manager == consts.USGovCloud_RMEndpoint:
+        cloud_name = consts.Azure_USGovCloudName
+    return cloud_name
```

Comment:
Why do you need to check the `resource_manager` value, instead of converting the `cloud_name` based on the original `cloud_name` value?

Reply:
updated the comparison to original cloud name